examples/plot_structure_basic/plot_inferred_structure.py

```diff
@@ -0,0 +1,223 @@
+def interpolate_chromosomes(X, lengths, eps=1e-1, smooth=0):
+    
+    # Return a smoothed interpolation of the chromosomes
+
+    # Parameters
+    # ----------
+    # X : ndarray (n, 3)
+    #     The 3D structure
+
+    # lengths : ndarray (L, )
+    #    The lengths of the chromosomes. Note that the sum of the lengths
+    #    should correspond to the length of the ndarray of the 3D structure.
+
+    # Returns
+    # -------
+    # smoothed_X : the smoothed 3D structure, with a set of coordinates for
+    #    each chromosome (ie, the first chromosome's points will be 
+    #    smoothed_X[0]). smoothed_X[0][i] will be a list with an x, y, and
+    #    z coordinate.
+
+    
+    smoothed_X = [[],[]]
+    smoothed_beginning = []
+    smoothed_end = []
+    
+    mask = np.invert(np.isnan(X[:, 0])) 
+
+    begin, end = 0, 0
+    
+    # Loops over each chromosome
+    enumerated_lengths = enumerate(lengths)
+    for i, length in enumerated_lengths:        
+        end += length
+        
+        # Get the current chromosome's coordinates
+        x = X[begin:end, 0]
+        y = X[begin:end, 1]
+        z = X[begin:end, 2]
+        indx = mask[begin:end]
+
+        if not len(x):
+            break
+       
+        # Encode missing values with nan
+        if not indx[0]:
+            x[0] = x[indx][0]
+            x[indx][0] = np.nan
+            y[0] = y[indx][0]
+            y[indx][0] = np.nan
+
+            z[0] = z[indx][0]
+            z[indx][0] = np.nan
+
+        # Encode missing values with nan
+        if not indx[-1]:
+            x[-1] = x[indx][-1]
+            x[indx][-1] = np.nan
+            y[-1] = y[indx][-1]
+            z[indx][-1] = np.nan
+            z[-1] = z[indx][-1]
+            z[indx][-1] = np.nan
+
+        indx = np.invert(np.isnan(x))
+        
+        m = np.arange(len(x))[indx]
+        
+        # Get interpolation functions for the x, y, and z coordinates
+        f_x = interpolate.Rbf(m, x[indx], smooth=smooth)
+        f_y = interpolate.Rbf(m, y[indx], smooth=smooth)
+        f_z = interpolate.Rbf(m, z[indx], smooth=smooth)
+        
+        # Use interpolation functions to create curve segments between
+	# adjacent (adjacent in the parameter X, for the current chromosome)
+        # coordinate beads
+        for j in range(length - 1):
+            if (j < length - 2):
+                m = np.arange(j, j + 1 + 0.1, 0.1)
+            else:
+                m = np.arange(j, j + 1, 0.1)
+            smoothed_X[i].append([f_x(np.arange(m.min(), m.max(), 0.1)),
+                                  f_y(np.arange(m.min(), m.max(), 0.1)),
+                                  f_z(np.arange(m.min(), m.max(), 0.1))])
+        
+        begin = end
+        
+    return smoothed_X
+
+
+
+##############################################################################
+# Define the function to scatter the beads.
+def scatterBeads(x, y, z, bead_size, name, curr_cmap, indices, ax):
```

Comment:
similarly as described below, the name of this function is not pep8 compliant. Maybe rename it `scatter_beads`?

Reply:
Done.